File: Python3/1018. Binary Prefix Divisible By 5.py
# 128ms, 95.12%; 14.8MB, 100.00%
class Solution:
    def prefixesDivBy5(self, A: list) -> list:
        res = [False] * len(A)
        if A[0] == 0: res[0] = True
        cur = A[0]
        for i in range(1, len(A)):
            cur = ((cur << 1) + A[i]) % 5
            if cur == 0: res[i] = True
        return res

File: 1001 - 1200/1018. Binary Prefix Divisible By 5.cc
// 12ms, 85.14%; 10.8MB, 78.57%
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<bool> prefixesDivBy5(const vector<int> &A) {
        int size = A.size();
        vector<bool> res;
        res.resize(size);
        if (A[0] == 0) res[0] = true;
        for (int i = 1, cur = A[0] % 5; i < size; ++i) {
            cur = ((cur << 1) + A[i]) % 5;
            if (cur == 0) res[i] = true;
        }
        return res;
    }
};
